### PR TITLE
A few new ports

### PR DESCRIPTION
**Pull request by Dennisbonke**
This PR adds 4 new ports that are listed as required or recommended dependencies of `OpenJDK` according to BLFS. Even if not needed for that, 3 of these ports are general nice to haves.

The following new ports were added:
- `cpio`,
- `giflib`,
- `unzip`,
- `zip`.

### CODE REVIEW

File: bootstrap.d/app-arch.yml [hunk 1]
```diff
@@ -190,6 +232,50 @@ packages:
...
+      patch-path-strip: 1
+      version: '6.0'
+    tools_required:
+      - system-gcc
+    pkgs_required:
+      - mlibc
+      - bzip2
+      - libiconv
+    configure:
+      # Fixup their shitty makefile, thanks gentoo!
```
Comment (avdgrinten) [inline bootstrap.d/app-arch.yml:258]: Remove the swear word / let's not blame other packages here.

**Review by avdgrinten — COMMENTED** [review]
Why does the unzip package come with a 16K patch file?

**Comment by Dennisbonke** [comment]
> Why does the unzip package come with a 16K patch file?

Courtesy of BLFS (which I should probably mention). It fixes a bunch of CVE's (listed in the patch) and some other issues. The last release of unzip has been a while. Also, from what I saw, the patch includes the unmodified source files as <name.c.orig>, inflating the size.

File: bootstrap.d/app-arch.yml (hunk 1 above)
Comment (Dennisbonke) [reply bootstrap.d/app-arch.yml]: You're right, my bad. Fixed it, still thanking gentoo for the sed as it is from there.

**Comment by avdgrinten** [comment]
This PR still has the 16K sloc patch file. Is that taken as-is from a reputable source? If yes, we should probably link to that source. Otherwise I'd be hesitant to merge this since we cannot validate the 16K sloc patch.

**Comment by Dennisbonke** [comment]
> This PR still has the 16K sloc patch file. Is that taken as-is from a reputable source? If yes, we should probably link to that source. Otherwise I'd be hesitant to merge this since we cannot validate the 16K sloc patch.

It was taken from [here](https://www.linuxfromscratch.org/blfs/view/svn/general/unzip.html), with a direct [download link](https://www.linuxfromscratch.org/patches/blfs/svn/unzip-6.0-consolidated_fixes-1.patch) available too. The submitter used an at linuxfromscratch.org and the patch was based on Fedora patches, as the patch itself notes. I deem this source trustable.

**Comment by avdgrinten** [comment]
Please add a link to the source to the patch's commit message.

**Comment by Dennisbonke** [comment]
> Please add a link to the source to the patch's commit message.

Done as requested.

**Pull request merged**